dynamicPermissions.groups.ps1

```diff
@@ -0,0 +1,456 @@
+            Headers = $headers
+            Method  = "GET"
+        }
+
+        $correlatedAccount = Invoke-HelloIDRestMethod @queryUserSplatParams
+    }
+    catch {
+        $ex = $PSItem
+        if ($($ex.Exception.GetType().FullName -eq "Microsoft.PowerShell.Commands.HttpResponseException") -or
+            $($ex.Exception.GetType().FullName -eq "System.Net.WebException")) {
+            $errorObj = Resolve-HelloIDError -ErrorObject $ex
+
+            $auditMessage = "Error querying account where [$($correlationField)] = [$($correlationValue)]. Error: $($errorObj.FriendlyMessage)"
+            Write-Warning "Error at Line [$($errorObj.ScriptLineNumber)]: $($errorObj.Line). Error: $($errorObj.ErrorDetails)"
+        }
+        else {
+            $auditMessage = "Error querying account where [$($correlationField)] = [$($correlationValue)]. Error: $($ex.Exception.Message)"
+            Write-Warning "Error at Line [$($ex.InvocationInfo.ScriptLineNumber)]: $($ex.InvocationInfo.Line). Error: $($ex.Exception.Message)"
+        }
+
+        $outputContext.AuditLogs.Add([PSCustomObject]@{
+                # Action  = "" # Optional
+                Message = $auditMessage
+                IsError = $true
+            })
+
+        # Throw terminal error
+        throw $auditMessage
+    }
+
+    # Compare desired with current permissions and grant permissions
+    foreach ($permission in $desiredPermissions.GetEnumerator()) {
+        $outputContext.SubPermissions.Add([PSCustomObject]@{
+                DisplayName = $permission.Value
+                Reference   = [PSCustomObject]@{ Id = $permission.Name }
+            })
+
+        if (-Not $currentPermissions.ContainsKey($permission.Name)) {
+            # Grant groupmembership
+            try {
+                $permissionBody = @{
+                    groupGuid = $permission.Name
+                }
+
+                $body = ($permissionBody | ConvertTo-Json -Depth 10)
+                $grantGroupMembershipSplatParams = @{
+                    Uri     = "$($actionContext.Configuration.baseUrl)/users/$($correlatedAccount.userGuid)/groups"
+                    Headers = $headers
+                    Method  = "POST"
+                    Body    = $body
+                }
+
+                if (-Not($actionContext.DryRun -eq $true)) {
+                    group: [$($permission.Value)] with groupGuid: [$($permission.Name)]
+                    Write-Verbose "Granting group: [$($permission.Value)] with groupGuid: [$($permission.Name)] to account with AccountReference: $($actionContext.References.Account | ConvertTo-Json)."
+                    Write-Verbose "Body: $($grantGroupMembershipSplatParams.Body)"
+
+                    $grantedGroupMembership = Invoke-HelloIDRestMethod @grantGroupMembershipSplatParams
+
+                    $outputContext.AuditLogs.Add([PSCustomObject]@{
+                            # Action  = "" # Optional
```

Comment:
When granting or revoking action is done in one script we should define the action with GrantPermission or RevokePermission.

Reply:
Agreed has been added.